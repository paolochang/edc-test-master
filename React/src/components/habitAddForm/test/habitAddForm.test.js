import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import HabitAddForm from "../habitAddForm";

describe("habitAddForm", () => {
  it("snapshot renders", () => {
    const component = render(<HabitAddForm onAdd={jest.fn()} />);
    expect(component.container).toMatchSnapshot();
  });

  describe("submit form", () => {
    let onAdd;
    let input;
    let button;
    beforeEach(() => {
      onAdd = jest.fn();
      render(<HabitAddForm onAdd={onAdd} />);
      input = screen.getByPlaceholderText("Habit");
      button = screen.getByText("Add");
    });

    it("calls onAdd when button is clicked and valid habit is entered", () => {
      userEvent.type(input, "New habit");
      userEvent.click(button);
      expect(onAdd).toHaveBeenCalledWith("New habit");
    });

    it("does not call onAdd when the habit is empty", () => {
      userEvent.type(input, "");
      userEvent.click(button);
      expect(onAdd).toHaveBeenCalledTimes(0);
    });
  });
});
