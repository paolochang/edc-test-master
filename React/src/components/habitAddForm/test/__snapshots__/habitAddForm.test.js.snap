// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`habitAddForm compares snapshot 1`] = `
<div>
  <form
    class="add-form"
  >
    <input
      class="add-input"
      placeholder="Habit"
      type="text"
    />
    <button
      class="add-button"
    >
      Add
    </button>
  </form>
</div>
`;
